src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -406,8 +425,24 @@ protected long fetchOffsetTokenWithRetry() {
                         event.getLastException().getMessage()))
             .build();
 
+    /**
+     * The fallback function to execute when all retries from getOffsetToken have exhausted.
+     * Fallback is only attempted on SFException
+     */
+    Fallback<Long> offsetTokenFallbackExecutor =
+        Fallback.builder(this::offsetTokenFallbackSupplier)
+            .handle(SFException.class)
+            .onFailure(
+                event ->
+                    LOGGER.error(
+                        "[OFFSET_TOKEN_FALLBACK] Failed to open Channel/fetch offsetToken for"
```

Comment:
Yes openChannel happens at two places:
1. Constructor of TopicPartitionChannel -> which means we will wipe of all states and it will call `precomputeOffsetTokenForChannel`
2. Failure handling which will call reopen, replace instance variable with new channel and call offsetToken.

Reply:
Then could you update the comment? Either adding reopen to it or only mentioning this function will be called as long as we need to fetch offset token would work

Reply:
Will do!